### PR TITLE
Added line number calculation to CompilerException, and made all current relevant exceptions use compilerexception

### PR DESCRIPTION
**Pull request by Mati-AAU**

### CODE REVIEW

File: GOAT-Compiler/Exceptions/CompilerException.cs [hunk 1]
```diff
@@ -8,16 +8,67 @@
...
+                // if we dont have a nodepositionvisitor, or this node doesnt belong to it, create a new one
+                if (nodePositionVisitor == null || !nodePositionVisitor.HasNode(node))
+                {
+                    RedoLineNumbers(node);
+                }
+                pos = nodePositionVisitor.GetPosition(node);
+            }
+            return "Compiler exception at " + pos;
+        }
+
```
Comment (thom776g) [inline GOAT-Compiler/Exceptions/CompilerException.cs:43]: Der er et enter for meget

File: GOAT-Compiler/Exceptions/FunctionAlreadyDefinedException.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using GOATCode.node;
+using System;
+
+namespace GOAT_Compiler.Exceptions
+{
+    public class FunctionAlreadyDefinedException : CompilerException
+    {
+        public FunctionAlreadyDefinedException(Node n, string Name) : base(n, $"The function named {Name} is already defined.")
+        {
+
```
Comment (thom776g) [inline GOAT-Compiler/Exceptions/FunctionAlreadyDefinedException.cs:10]: eNtEr

File: GOAT-Compiler/Exceptions/RefNotFoundException.cs [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 using System.ComponentModel;
 
 namespace GOAT_Compiler.Exceptions
 {
-    public class RefNotFoundException : Exception
+    public class RefNotFoundException : CompilerException
     {
-        public RefNotFoundException(string SymbolName) : base($"Symbol named: {SymbolName} not found")
+        public RefNotFoundException(Node n, string SymbolName) : base(n, $"Symbol named: {SymbolName} not found")
         {
```
Comment (thom776g) [inline GOAT-Compiler/Exceptions/RefNotFoundException.cs:11]: again with the enter

File: GOAT-Compiler/Exceptions/CompilerException.cs (hunk 1 above)
Comment (Mati-AAU) [reply GOAT-Compiler/Exceptions/CompilerException.cs:43]: done

File: GOAT-Compiler/Exceptions/FunctionAlreadyDefinedException.cs (hunk 2 above)
Comment (Mati-AAU) [reply GOAT-Compiler/Exceptions/FunctionAlreadyDefinedException.cs:10]: done

File: GOAT-Compiler/Exceptions/RefNotFoundException.cs (hunk 3 above)
Comment (Mati-AAU) [reply GOAT-Compiler/Exceptions/RefNotFoundException.cs:11]: done

**Review by thom776g — APPROVED** [review]

**Pull request merged**